src/Compilers/Core/RebuildTest/RoundTripUtil.cs

```diff
@@ -86,14 +91,118 @@ public static void VerifyRoundTrip<TCompilation>(TCompilation original)
             VerifyCompilationOptions(original.Options, rebuild.Options);
 
             using var rebuildStream = new MemoryStream();
+            using var rebuildPdbStream = optionsReader.HasEmbeddedPdb ? null : new MemoryStream();
             var result = compilationFactory.Emit(
                 rebuildStream,
+                rebuildPdbStream,
                 rebuild,
                 embeddedTexts: ImmutableArray<EmbeddedText>.Empty,
                 CancellationToken.None);
             Assert.Empty(result.Diagnostics);
             Assert.True(result.Success);
-            Assert.Equal(originalBytes.ToArray(), rebuildStream.ToArray());
+
+            var rebuildBytes = rebuildStream.ToImmutable();
+            var rebuildReader = new PEReader(rebuildBytes);
+
+            ImmutableArray<byte> rebuildPdbBytes;
+            MetadataReader rebuildPdbReader;
+            if (rebuildPdbStream is null)
+            {
+                rebuildPdbReader = rebuildReader.GetEmbeddedPdbMetadataReader() ?? throw new InvalidOperationException();
+                rebuildPdbBytes = new ReadOnlySpan<byte>(rebuildPdbReader.MetadataPointer, rebuildPdbReader.MetadataLength).ToArray().ToImmutableArray();
```

Comment:
>ToArray() [](start = 124, length = 9)

Minor: Is `ToArray()` needed?

Reply:
Unfortunately, I haven't been able to find API on ReadOnlySpan which allows creating an ImmutableArray without copying (save for creating a builder and copying the items over manually, I guess, which we could do)

https://stackoverflow.com/questions/66912187/how-should-i-create-an-immutablearray-from-a-span-readonlyspan?noredirect=1#comment118278058_66912187

Reply:
Punting on this since it's test code.